src/library_dylink.js

```diff
@@ -824,6 +824,20 @@ var LibraryDylink = {
   },
 #endif
 
+  $newDSO: function(name, handle, syms) {
+    var dso = {
+      refcount: Infinity,
+      name: name,
+      module: syms,
+      global: true,
+    };
+    LDSO.loadedLibsByName[name] = dso;
+    if (handle != undefined) {
```

Comment:
When is the handle undefined? It would be simpler to reason about if the handle could always be assumed to exist.

Reply:
The then DSO are loaded during the JS start process, before native code run, we load all any needed shared libraries by name, but these don't have any native "dso" struct/handler associated with them.    They get handles don't someone calls "dlopen" on them... which is often never in the case of load time dynamic libraries.

Reply:
Makes sense, thanks!